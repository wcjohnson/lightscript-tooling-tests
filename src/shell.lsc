exec = require('child_process').exec

export run(cmd) ->
  new Promise! (resolve, reject) ->
    exec! cmd, (err, stdout, stderr) ->
      if err:
        reject({err, stdout, stderr})
      else:
        resolve({stdout, stderr})

export snapshot(expect, cmd) ->
  run! cmd
    .then! ({stdout, stderr}) ->
      expect({stdout, stderr, succeeded: true, err: null}).toMatchSnapshot()
    .catch! ({err, stdout, stderr}) ->
      expect({stdout, stderr, succeeded: false, err}).toMatchSnapshot()

export successfulSnapshot(expect, cmd) ->
  run! cmd
    .then! ({stdout, stderr}) ->
      expect({stdout, stderr, succeeded: true, err: null}).toMatchSnapshot()
