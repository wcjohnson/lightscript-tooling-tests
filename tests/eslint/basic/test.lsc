import { snapshot } from '../../../lib/shell'

test! "lint self", ->
  expect~snapshot! "$(npm bin)/eslint tests/eslint/basic/lintThis.lsc"
