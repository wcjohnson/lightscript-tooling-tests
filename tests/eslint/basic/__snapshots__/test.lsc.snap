// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint self 1`] = `
Object {
  "exitStatus": 1,
  "stderr": "",
  "stdout": "
/Users/wcj/dev/lightscript/lightscript-tooling-tests/tests/eslint/basic/lintThis.lsc
  1:1  error  'x' is not defined  no-undef

âœ– 1 problem (1 error, 0 warnings)

",
  "succeeded": false,
}
`;
